### ISSUE TITLE
[popups] Implement on_hover hook

### ISSUE THREAD

**Issue by FichteFoll** [SublimeLinter/SublimeLinter3#471]
In a similar vein to https://github.com/SublimeLinter/SublimeLinter3/issues/470 (and much more to my preference), it would be nice if hover events were interpreted to show the error description in a popup.

Because of the many possible ways to show context information now, I suggest adding a setting that specifies which locations are active to allow multiple. (This is [SublimeCodeIntel's method](https://github.com/SublimeCodeIntel/SublimeCodeIntel/blob/b062430a1329f51e7c6ca22fd7de699635ead607/SublimeCodeIntel.sublime-settings#L52-L59) on steroids.)
Since hover events can happen on text or the gutter, this should be a separate option.
And, furthermore, text could be shown either when hovering anywhere on the line or only when the highlighted region (column) is hovered.

Example:

``` json
"error_message_location": ["statusbar", "hover_gutter", "hover_line", "hover_region", "phantom"]
```

where the current behavior would be equal to `["statusbar"]`.

**Comment by buildersbrewery** [comment]
related to PR: SublimeLinter/SublimeLinter3#318

**Comment by FichteFoll** [comment]
Thanks for the reference. That could also be an option for showing messages imo although it could collide with other hover styles. Maybe as `"line_popup"`.

**Comment by groteworld** [comment]
Now that #510 is nearing merge, I believe on_hover is the next logical step for this. Thank you for adding on this conversation.

**Comment by FichteFoll** [comment]
Closing since this has been implemented nicely in #666.
